### PR TITLE
Add support for Action Workflow state updates

### LINKED ISSUE #1637: Add Workflow enable and disable states
Github has two APIs to manually enable and disable a workflow by ID or Filename,

- [Enable a workflow](https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#enable-a-workflow)
- [Disable a workflow](https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow)

Both are not implemented yet. I would like to take this up if it is supposed to be added.

### PR DESCRIPTION
**Pull request by n1lesh**
Fixes #1637  

Includes enable/disable by ID and Filename w/ tests.

### CODE REVIEW

File: github/actions_workflows.go [hunk 1]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#enable-a-workflow
+func (s *ActionsService) EnableWorkflowByID(ctx context.Context, owner, repo string, workflowID int64) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/workflows/%v/enable", owner, repo, workflowID)
+	return s.enableWorkflow(ctx, u)
+}
+
+// EnableWorkflowByFileName enables a workflow and sets the state of the workflow to active
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#enable-a-workflow
```
Comment (gmlewis) [inline github/actions_workflows.go:176]:
```suggestion
// GitHub API docs: https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#enable-a-workflow
```

File: github/actions_workflows.go [hunk 2]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(ctx, req, nil)
+}
+
+// DisableWorkflowByID disables a workflow and sets the state of the workflow to disabled_manually
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
```
Comment (gmlewis) [inline github/actions_workflows.go:193]:
```suggestion
// GitHub API docs: https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
```

File: github/actions_workflows.go [hunk 3]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+func (s *ActionsService) DisableWorkflowByID(ctx context.Context, owner, repo string, workflowID int64) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/workflows/%v/disable", owner, repo, workflowID)
+	return s.enableWorkflow(ctx, u)
+}
+
+// DisableWorkflowByFileName disables a workflow and sets the state of the workflow to disabled_manually
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
```
Comment (gmlewis) [inline github/actions_workflows.go:201]:
```suggestion
// GitHub API docs: https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
```

File: github/actions_workflows.go [hunk 4]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+func (s *ActionsService) enableWorkflow(ctx context.Context, url string) (*Response, error) {
+	req, err := s.client.NewRequest("PUT", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(ctx, req, nil)
+}
+
+// DisableWorkflowByID disables a workflow and sets the state of the workflow to disabled_manually
```
Comment (gmlewis) [inline github/actions_workflows.go:191]:
```suggestion
// DisableWorkflowByID disables a workflow and sets the state of the workflow to "disabled_manually".
```

File: github/actions_workflows.go [hunk 5]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
+func (s *ActionsService) DisableWorkflowByID(ctx context.Context, owner, repo string, workflowID int64) (*Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/actions/workflows/%v/disable", owner, repo, workflowID)
+	return s.enableWorkflow(ctx, u)
+}
+
+// DisableWorkflowByFileName disables a workflow and sets the state of the workflow to disabled_manually
```
Comment (gmlewis) [inline github/actions_workflows.go:199]:
```suggestion
// DisableWorkflowByFileName disables a workflow and sets the state of the workflow to "disabled_manually".
```

File: github/actions_workflows.go [hunk 6]
```diff
@@ -162,3 +162,53 @@ func (s *ActionsService) CreateWorkflowDispatchEvent(ctx context.Context, owner,
...
+//
+// https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#disable-a-workflow
+func (s *ActionsService) DisableWorkflowByFileName(ctx context.Context, owner, repo string, workflowFileName string) (*Response, error) {
```
Comment (gmlewis) [inline github/actions_workflows.go:202]:
```suggestion
func (s *ActionsService) DisableWorkflowByFileName(ctx context.Context, owner, repo, workflowFileName string) (*Response, error) {
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
This looks great, @n1lesh - just a few minor tweaks, please... mainly to help keep the auto-generated godocs to read nicely and be consistent with the rest of the repo.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Please revert the addition of ".DS_Store" to this PR.
In fact, if you want to add ".DS_Store" to the ".gitignore" file in this PR instead, that would prevent this from happening again.  Thanks.

**Comment by n1lesh** [comment]
> Please revert the addition of ".DS_Store" to this PR.
> In fact, if you want to add ".DS_Store" to the ".gitignore" file in this PR instead, that would prevent this from happening again. Thanks.

Should have noticed this before committing. My apologies!

**Review by gmlewis — APPROVED** [review]
> Should have noticed this before committing. My apologies!

No problem!  Thank you, @n1lesh !
LGTM.

Awaiting second LGTM before merging.

**Review by nightlark — APPROVED** [review]
LGTM

**Comment by gmlewis** [comment]
Thank you, @nightlark !
Merging.

**Pull request merged**